edsdme/blocks/announcements-preview/announcements-preview.js

```diff
@@ -0,0 +1,147 @@
+    linkWrapper.target = '_blank';
+
+    linkWrapper.style.display = 'block';
+
+    const announcementItem = document.createElement('div');
+    announcementItem.className = 'announcement-item';
+
+    const contentWrapper = document.createElement('div');
+    contentWrapper.className = 'card-content';
+    const titleElement = document.createElement('h3');
+    titleElement.className = 'card-title';
+    titleElement.textContent = cardData.contentArea.title;
+
+    const descriptionElement = document.createElement('p');
+    descriptionElement.className = 'card-description';
+    descriptionElement.textContent = cardData.contentArea.description;
+
+    const imageWrapper = document.createElement('div');
+    imageWrapper.className = 'card-image';
+
+    const picture = document.createElement('picture');
+
+    const source = document.createElement('source');
+    source.srcset = `${new URL(cardData.styles?.backgroundImage).pathname}?width=240&format=webp&optimize=small`;
+    source.type = 'image/webp';
+
+    const img = document.createElement('img');
+    img.src = `${new URL(cardData.styles?.backgroundImage).pathname}??width=240&format=webp&optimize=small`;
+    img.alt = 'Image description';
+    img.loading = 'lazy';
+
+    picture.appendChild(source);
+    picture.appendChild(img);
+
+    imageWrapper.appendChild(picture);
+    announcementItem.appendChild(imageWrapper);
+    contentWrapper.appendChild(titleElement);
+    contentWrapper.appendChild(descriptionElement);
+    announcementItem.appendChild(contentWrapper);
+
+    linkWrapper.appendChild(announcementItem);
+    app.appendChild(linkWrapper);
+  }
+
+  async function fetchData() {
+    try {
+      let apiData = [];
+
+      const response = await fetch(blockData.caasUrl);
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      apiData = await response.json();
+
+      if (apiData?.cards) {
+        apiData.cards.forEach((card) => {
+          const cardDate = new Date(card.cardDate);
+          if (newestCards.length < 3) {
+            newestCards.push(card);
+            newestCards.sort((a, b) => new Date(b.cardDate) - new Date(a.cardDate));
+          } else if (cardDate > new Date(newestCards[2].cardDate)) {
+            newestCards[2] = card;
+            newestCards.sort((a, b) => new Date(b.cardDate) - new Date(a.cardDate));
+          }
+        });
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error('Error fetching data:', error);
+    }
+  }
+
+  await fetchData();
+
+  if (blockData.title !== '') {
+    const componentTitle = document.createElement('div');
+    componentTitle.className = 'text announcement-preview-title';
+    const titleText = document.createElement('h3');
+    titleText.textContent = blockData.title;
+    componentTitle.appendChild(titleText);
+    app.appendChild(componentTitle);
+  }
+
+  if (newestCards.length !== 0) {
```

Comment:
Here you can just check for `newestCards.length` if it's 0 it will result in false.